### PR TITLE
[Merged by Bors] - chore: import the `header` linter in all files imported in Mathlib.Init

### PR DESCRIPTION
**Pull request by grunweg**
Every file importing Mathlib.Init transitively imports this file already, thus having its module docstring and copyright header checked by the `header` linter. This adds the same input to every file *imported in* Mathlib.Init.

Helps with #17556.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by adomani** [comment]
Since I doubt that the `Header` import is going to be used for importing any function other than the linter, maybe it makes sense to add the file to the part of shake where the files that are globally ignored go, so that then it does not need to be shaken out in every individual import.

**Comment by grunweg** [comment]
I don't feel strongly about this - if you think this is useful, I'm happy to do that instead.

**Comment by grunweg** [comment]
Actually: yes, that seems like a nice simplification - and #18281 will flag any new files which import just the Header linter, so this is not likely to regress unintendedly. Changed.

File: Mathlib/Tactic/Finiteness/Attr.lean [hunk 1]
```diff
@@ -4,6 +4,7 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Floris van Doorn
 -/
 import Aesop
+import Mathlib.Init
```
Comment (adomani) [inline Mathlib/Tactic/Finiteness/Attr.lean:7]: This means that all of `aesop` will be imported in `Init`: maybe there should be a wider discussion about this.

(In fact, I'm not sure that `aesop` has not already been imported with `finiteness`, actually.)

Comment (grunweg) [reply Mathlib/Tactic/Finiteness/Attr.lean:7]: I can confirm; I think you have it packwards: this file imports Mathlib.Init (now), and it imports aesop (and did before this PR).

**Comment by adomani** [comment]
Thanks!

maintainer merge

**Review by Vierkantor — APPROVED** [review]
Thanks!

bors r+

**Comment by grunweg** [comment]
Thanks for the reviews!

**Pull request closed without merging**